src/GafferScene/Instancer.cpp

```diff
@@ -323,7 +337,103 @@ class Instancer::EngineData : public Data
 			}
 		}
 
+		void initPrototypes( PrototypeMode &mode, const std::string &index, const std::string &rootsVariable, const StringVectorData *rootsList )
+		{
+			const std::vector<std::string> *rootStrings = nullptr;
```

Comment:
For this "declare variable and then initialise it differently in different branches" pattern, I've taken to _not_ initialising the variable at the point of declaration. That way the compiler will tell you if you miss an initialisation lower down and then try to use it uninitialised.

Reply:
Addressed in e19c79dbb000bed20de0a249fe3d7ee44d2d0a6d, though to be honest, I'm not sure I like it. I had to add a nonsensical `default` case to satisfy the compiler.

Reply:
Interesting. GCC 6 doesn't seem to want the extra `case` - I guess you're using 4.8? But GCC 6 doesn't detect the uninitialised variable if I omit it from one of the case statements either, so obviously I've been trusting the compiler too much. Revert it then I guess...

Reply:
Addressed during the squash rebase